### ISSUE TITLE
Google APIException (10): Unable to validate key file signature: no public key found

### ISSUE THREAD

**Issue by eriktews** [corona-warn-app/cwa-app-android#817]
## Avoid duplicates
* [-] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x] Bug is not already reported in another issue

There is something very generic in the FAQ, but there is no issue on GitHub that is open and mentions this specific exception.

## Describe the bug

When I open the app, I get the message: _URSACHE: 3 Etwas ist schief gelaufen. Fehler bei der Kommunikation mit der Google API (10)._

When I then click on details, I can see a stack backtrace that starts with _Unable to validate key file signature: no public key found_.

I installed the app on the first day of the public release and since then it was continuously active and never showed a warning.

## Expected behaviour

It should just start and state that the app is working fine and tell me my personal risk.

## Steps to reproduce the issue

Here, it's sufficient to just open the app. Attached is also a screenshot.

![screenshot](https://user-images.githubusercontent.com/1436802/86528925-f0185400-beac-11ea-8cbc-da20d98ef0b3.jpg)

## Technical details

- Mobile device: Pixel 3
- Android version: Android 10 with June security updates.

## Possible Fix

I have no idea.

## Additional context

I did nothing odd with the phone, no other contact tracing app is installed and the device is not rooted or something like that.

---
Internal Tracking ID: EXPOSUREAPP-1714

**Comment by eriktews** [comment]
Hi, and in case you are interested, I can attach an USB cable to the phone and run some adb commands.

**Comment by marci4** [comment]
Hey,
I have the same issue on a Huawei P6
Running playstore version 20.7.16-all [0] [PR] 317546459 and android 6.0

Best regards,
Marcel

**Comment by ggaussling** [comment]
This is a duplicate of the closed bug #737 https://github.com/corona-warn-app/cwa-app-android/issues/737#issuecomment-653630609

I don't know how complete the work on this bug is. I'm affected too, and receive no update of my risk state since 2020/06/29, due to this bug. I think it's only affecting Android 6.x smartphones yet.

The reason is identified and the developers are waiting on reaction of google to roll out an update of the culprit google play API.

**Comment by eriktews** [comment]
> This is a duplicate of the closed bug #737 [#737 (comment)](https://github.com/corona-warn-app/cwa-app-android/issues/737#issuecomment-653630609)
> 
> I don't know how complete the work on this bug is. I'm affected too, and receive no update of my risk state since 2020/06/29, due to this bug. I think it's only affecting Android 6.x smartphones yet.

I'm don't think it's the same as #737, first of all, there the error message is slightly different: _Unable to validate key file signature: Pipe is closed_, and second this bug seems to be about problems on Android 6.x, but I am using the app on an Android 10 device. Still, the issues might be related.

**Comment by eriktews** [comment]
> I have the same issue on a Huawei P6
> Running playstore version 20.7.16-all [0] [PR] 317546459 and android 6.0

Can you check whether it's really the same exception you are seeing there, or is the stack backtrace slightly different? (Take a look at the screenshot I posted)

**Comment by treysis** [comment]
Which version of PlayServices?

**Comment by eriktews** [comment]
> Which version of PlayServices?

That's 20.21.17, and there is no further update available in the play store.

**Comment by eriktews** [comment]
My app just returned back to normal, but shows that it's active 13 out of 14 days, but it's definitely active for a longer time and the phone was never switched off for a full day or so. So as long as the error doesn't return, I will probably not be able to support the debugging here.

**Comment by fjgabler** [comment]
Same issue here! Before I had issue #699 with error "Nearby.EXPOSURE_NOTIFICATION_API is not avaliabe on this device" and could not activate "Riskio-Ermittlung". Since the last update of Google Play Services I was able to activate "Risiko-Ermittlung" but get the error reported by @eriktews.

YOTA 3
Android 7.1.1
Google Play Services 20.24.14
Corona-Warn 1.0.5

**Comment by Random-4** [comment]
Hi,
I have seen the Corona APP working fine for the first days / weeks. 
Isn't it possible to downgrade the mobile device to a level whereitwas workingfine?
(What specific component would thatbe? OS or Browser?)
This should be a workaround until Google managed to solve the problem. 
After a solution is published I would be fine to load again the latest version. 

Thx for your help. 
I'd like to

**Comment by treysis** [comment]
Is this really google? SwissCovid doesnt have this issue.

**Comment by tkowark** [comment]
Yes, it does: https://github.com/DP-3T/dp3t-app-android-ch/issues/100

**Comment by tkowark** [comment]
It does have error 10 (https://github.com/DP-3T/dp3t-app-android-ch/issues/133), but since the full stack trace is not dumped, it's hard to tell that it is not the same error?

**Comment by frankcom** [comment]
So it needs to be fixed with Google?

**Comment by vaubaehn** [comment]
> 
> 
> So it needs to be fixed with Google?

Yes. Some hours ago, a new version of Google Play Services was released: 20.24.65. If they managed to fix current API problems in that version, within the next 2 weeks our devices should receive it, and CWA should function properly again. If the fixes are delayed to a later version of Play Services, then it may take some weeks longer...

**Comment by frankcom** [comment]
Thanks for the explanation - hope Google has implemented it already.

**Comment by vaubaehn** [comment]
@frankcom My information about version 20.24.65 was taken from available versions of Google Play Services on apkmirror (https://www.apkmirror.com/apk/google-inc/google-play-services/). But obviously it's not that reliable to track versions from there, as my smartphone received an automatic update from 20.24.14 to 20.26.14 (skipping 20.24.65) pushed from Google last Friday, even I am not participating in the beta-program.
And, unfortunately, for me problems (API 10 and API 39508 on Android 6) have not yet been solved with GPS 20.26.14....

**Comment by xyxyxyxy49** [comment]
I've got the same error here. Till yesterday it worked like a charm, but now I get the mentioned error. 

CWA 1.0.5

Huawei P20 Pro
Android 10.0.0.161
GPS 20.24.14 (last update 3.7.20)
Play store 21.0.17-all 

Since there were no updates for GPS in the last days, why did the error occur just today.

Can do test if wanted, have everything here.

**Comment by xyxyxyxy49** [comment]
Just a little followup on my problem. After I got an update for Google play services the error vanished.
I'm now using GPS 20.26.14, and again, no error anymore.

**Comment by SebastianWolf-SAP** [comment]
@xyxyxyxy49 Great news and thanks for your reply. Could you please check and confirm that you are running ENF version 1.5? See https://www.coronawarn.app/en/faq/#ENF_version on how to determine if you are running version 1.5.

**Comment by xyxyxyxy49** [comment]
![Screenshot_20200722_151328_com google android gms](https://user-images.githubusercontent.com/28436752/88180650-f8eb8280-cc2d-11ea-96c6-75c336b99cb4.jpg)

Seems like it's running ENF in version 1.5

**Comment by vaubaehn** [comment]
@SebastianWolf-SAP @xyxyxyxy49 So, ENF 1.5 apparently may solve 'API 10/no public key found' for Android 10. Let's hope the best then also for 'API 10/pipe closed' for Android 6.

**Comment by SebastianWolf-SAP** [comment]
@vaubaehn Well, that was exactly the information which we received from Google, this issue here should be gone with ENF 1.5. But we would like to be 100% sure about it before we communicate this also to others. ;)

Therefore: Everybody who has this error 10 issue, please update Google Play issues, check for ENF version 1.5 and report back if you still face the issue or if things are better now.

**Comment by eriktews** [comment]
So at least here the error is gone now.

But maybe one suggestion about the app: When you go to the "App-Informationen" screen, it just shows the version number of the app itself. Could you include the GPS and ENF version numbers there as well? Maybe that will make debugging such issues much easier in the future.

**Comment by Tj20** [comment]
> Let's hope the best then also for 'API 10/pipe closed' for Android 6.

The appropriate bug report thread #737 is still closed, but the 'API 10/pipe closed' issue seems to have been fixed for me on my Android 6 phone. Nice.

(I'm using: ENF version 1.5 (1520292003), Android 6.0, Google Play Services 20.24.14, Google Play Store 20.9.10-all)

**Comment by tkowark** [comment]
Thank you all again for your input. As we receive plenty confirmation that the error is gone with the latest Exposure Notification Framework version (https://www.coronawarn.app/de/faq/#ENF_version), we will accordingly close this issue.

If you still experience the issue **despite** running on the latest ENF version, please open a new issue in this repository!

**Comment by dsarkar** [comment]
Please, see https://github.com/corona-warn-app/cwa-app-android/issues/1645#issuecomment-736297560.
